src/sysupdate/sysupdate-transfer.c

```diff
@@ -466,18 +460,42 @@ static int config_parse_partition_flags(
         return 0;
 }
 
-int transfer_read_definition(Transfer *t, const char *path) {
-        int r;
+static bool decide_if_enabled(Hashmap *known, char **regular, char **requisite) {
```

Comment:
as suggested below, i think the two arrays should be part of the Transfer object, so that this function can then just take that as first argument. And maybe name the function transfer_decide_if_enabled() or so.